batcher/aligned-batcher/src/lib.rs

```diff
@@ -1083,7 +1090,6 @@ impl Batcher {
         signatures: Vec<SignatureData>,
         fee_params: CreateNewTaskFeeParams,
     ) -> Result<TransactionReceipt, BatcherError> {
-        // pad leaves to next power of 2
         let padded_leaves = Self::pad_leaves(leaves);
```

Comment:
Removed comment but still padding the leaves.

The idea is to remove the padding? If the padding is removed, the contract should have a change as well

Reply:
Addressed! Removing the padding lead to Tx reversion in the BatcherPaymentService due to an InvalidMerkleRoot. The comment above goes into more detail.